### PR TITLE
Add new features

### PR DESCRIPTION
**Pull request by riittagerhards**
- [x] Show text in gallery card if no image for the day available

image

- [x] Add patch in index.ts and test it with samples.http to prepare edit page
- [x] Add edit page

image

- [x] Add an empty page if user wants to edit a non existing day
- [x] Improve styling on gallery page (responsive design for gallery cards).

### CODE REVIEW

File: src/app/pages/EditEntry/EditEntry.tsx [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+            placeholder={entry?.title}
+            value={title}
+            onChange={(event) => setTitle(event.target.value)}
+            required
+          />
+          <input
+            className={styles.dateInput}
+            type="date"
+            value={date}
+            readOnly
```
Comment (manusmd) [inline src/app/pages/EditEntry/EditEntry.tsx:87]: Nice! Didn't know that this keyword exists 👍

File: src/app/pages/SingleEntry/SingleEntry.tsx [hunk 2]
```diff
@@ -63,7 +63,16 @@ function SingleEntry(): JSX.Element {
...
-          <Button name="delete" type="button" onClick={() => handleClick()} />
+          <span className={styles.buttonContainer}>
+            <Link
+              key={entry.date}
+              to={`/edit/${entry.date}`}
+              className={styles.edit}
+            >
+              Edit
+            </Link>
+            <Button name="Delete" type="button" onClick={() => handleClick()} />
```
Comment (manusmd) [inline src/app/pages/SingleEntry/SingleEntry.tsx:74]: I guess the name is used for the text of the button, is it? I would suggest:
- (if you haven't done it yet) giving the name in your button component the type:
`name: 'delete' | //other things you use your button for`
- writing 'delete' in lowercase
- and then use CSS to type the first letter of the button text in uppercase

**Review by manusmd — APPROVED** [review]
Just a few optional things. Code looks very clean! Nice work 🚀

File: src/app/pages/EditEntry/EditEntry.module.css [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+
+@keyframes fadein {
+  from {
+    opacity: 0.5;
+  }
+
+  to {
+    opacity: 1;
+  }
+}
```
[the remark this answers is not in the document]
Comment (riittagerhards) [reply src/app/pages/EditEntry/EditEntry.module.css:13]: I tried it, this one works better I think.

File: src/app/pages/EditEntry/EditEntry.tsx [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+    };
+
+    try {
+      await editEntry(newEntry);
+    } catch (error) {
+      console.log(error);
+    }
+    setTimeout(() => {
+      navigate('/gallery');
+    }, 500);
```
[the remark this answers is not in the document]
Comment (riittagerhards) [reply src/app/pages/EditEntry/EditEntry.tsx:55]: I don't really need it. But I want the effect that the page doesn't change too fast.

File: src/app/pages/EditEntry/EditEntry.tsx [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+      title,
+      date,
+      text,
+    };
+
+    try {
+      await editEntry(newEntry);
+    } catch (error) {
+      console.log(error);
+    }
```
[the remark this answers is not in the document]
Comment (riittagerhards) [reply src/app/pages/EditEntry/EditEntry.tsx:52]: Good idea!

File: src/app/pages/SingleEntry/SingleEntry.tsx (hunk 2 above)
Comment (riittagerhards) [reply src/app/pages/SingleEntry/SingleEntry.tsx:74]: 👍

**Pull request merged**